# Supervised learning- Logistic Regression model 

1) For deploying a logistic rgeression model, following packages should be installed and imported

In [12]:
import pandas as pd
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split

2) Get data and describe features and target variables

In [17]:
diabetes= pd.read_csv(r"C:\Users\Pabba Abhishek\Desktop\MSRF\diabetes.csv")
print(diabetes.head())
feature_cols = ['pregnant', 'insulin', 'bmi', 'age','glucose','bp','pedigree']
x = diabetes[feature_cols] # Features
y = diabetes.label # Target variable

   pregnant  glucose  bp  skin  insulin   bmi  pedigree  age  label
0         6      148  72    35        0  33.6     0.627   50      1
1         1       85  66    29        0  26.6     0.351   31      0
2         8      183  64     0        0  23.3     0.672   32      1
3         1       89  66    23       94  28.1     0.167   21      0
4         0      137  40    35      168  43.1     2.288   33      1


3) Splitting the datasets for testing and training

In [14]:
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.25,random_state=0)

4) Create a model and train it with the data

In [15]:
model=LogisticRegression()
model.fit(x_train,y_train)

LogisticRegression()

5) Evaluate the model:

In [16]:
model.score(X_test,y_test)

0.8072916666666666

The above code is an example logistic regression model deployed using scikit-learn. Being supervised learning, logistic regression model requires features and target columns, here given as x and y. To check the accuracy of the prediction, the data is split into test and train and the train data is used to train the model. Then, the test data is used to find the prediction accuracy. Here, the score is ~81% indicating, the prediction is ~81% accurate.